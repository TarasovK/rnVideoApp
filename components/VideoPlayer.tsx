import React from 'react'
import { StyleSheet } from 'react-native'
import Video from 'react-native-video'

export interface Props {
    uri: string
}

interface State {
    paused: boolean
}


export class VideoPlayer extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props)   
        this.state = {
            paused: true
        }     
    }

    play = () => {
        this.setState( {
            paused: false
        })
    }

    pause = () => {
        this.setState( {
            paused: true
        })
    }
    
    render() {       
        return (
            <Video 
                source = {{uri: this.props.uri}}             
                style = {styles.backgroundVideo} 
                resizeMode = "contain"
                repeat = {true}
                paused={this.state.paused}/>
        );
    }
}

var styles = StyleSheet.create({
    backgroundVideo: {
      position: 'absolute',
      top: 0,
      left: 0,
      bottom: 0,
      right: 0,
    },
  });