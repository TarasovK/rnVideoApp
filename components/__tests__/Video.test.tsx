import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {VideoPlayer} from '../VideoPlayer';

jest.mock('react-native-video', () => 'Video');

describe('VideoPlayer', () => {
  const uri =
    'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4';
  it('renders video correctly', () => {
    const component = renderer
      .create(<VideoPlayer uri={uri}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});