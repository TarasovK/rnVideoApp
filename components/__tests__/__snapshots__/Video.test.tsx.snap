// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer renders video correctly 1`] = `
<Video
  paused={true}
  repeat={true}
  resizeMode="contain"
  source={
    Object {
      "uri": "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4",
    }
  }
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
